app/controllers/users_controller.rb

```diff
@@ -209,6 +209,49 @@ def check_emails
     render json: failed_json, status: 403
   end
 
+  def update_primary_email
```

Comment:
I wonder if we should add an entry in the staff logs about this operation?

Reply:
That sounds like a good idea.